gap/attr.gi

```diff
@@ -3358,6 +3358,157 @@ function(D)
+  # longer a (u, v)-path.
+
+  # Let u and v be non-adjacent vertices in `digraph`. Because of the
+  # correspondence of paths in `digraph` and `doubled_digraph`, any such set S
+  # for `digraph` corresponds to a set of edges E_S (obtained by replacing the
+  # vertex w by the edge (2*w-1, 2*w)) in `doubled_digraph` whose removal
+  # disconnects 2*u from 2*v-1.
+  # Conversely, it can be shown that the smallest set of edges disconnecting
+  # 2*u from 2*v-1 has the same cardinality as E_S. This is because, whenever
+  # we remove any edge (2*s, 2*t-1) from `doubled_digraph` with the goal of
+  # disconnecting 2*u from 2*v-1, it is always just as
+  # efficient or more efficient to remove the edge (2*s-1, 2*s) instead, since
+  # any path from 2*u to 2*v-1 utilizing (2*s, 2*t-1) in `doubled_digraph` must
+  # pass through (2*s-1, 2*s) by construction. Note that u and v are
+  # non-adjacent by assumption, so it cannot be the case that
+  # (2*s, 2*t-1) = (2*u, 2*v-1).
+  # It follows that, for non-adjacent vertices, the local connectivity of u and
+  # v in `digraph` is equal to the minimum cut with source 2*u-1 and target 2*v
+  # in `doubled_digraph`.
+
+  # By the max-flow min-cut theorem, the size of the minimum cut with source u
+  # and target v equals the maximum flow between u and v see Wikipedia below:
+  # https://en.wikipedia.org/wiki/Max-flow_min-cut_theorem
+  # Hence we can compute local vertex connectivity by repeated calls to the
+  # max_flow function below:
+  max_flow := {D, source, target} ->
+    Sum(DigraphMaximumFlow(D, source, target)[source]);
+
+  # The vertex connectivity is the minimum of the local vertex connectivity
+  # over all pairs of vertices. However, it can be computed a bit more
+  # cleverly by utilizing some theory to reduce the number of pairs considered.
+  # In this function we implement Algorithm 11 from Abdol-Hossein
+  # Esfahanian's ``Connectivity Algorithms'' which can be found at
+  # https://www.cse.msu.edu/~cse835/Papers/Graph_connectivity_revised.pdf
+  # In particular, we reduce the number of local vertex connectivity
+  # computations to n-d-1 + d*(d-1)/2 where n is the total number of vertices
+  # and d is the minimum degree of any vertex.
+  v := PositionMinimum(OutDegrees(digraph));
+  neighbours_v := OutNeighboursOfVertex(digraph, v);
+  # TODO: is adjacency function the best way to check adjacency?
+  adjacency_function := DigraphAdjacencyFunction(digraph);
+  kappa_min := fail;
+  for u in DigraphVertices(digraph) do
+    if u <> v and not adjacency_function(v, u) then
+      kappa := max_flow(doubled_digraph, 2 * u, 2 * v - 1);
+      if kappa_min = fail or kappa < kappa_min then
+        kappa_min := kappa;
+      fi;
+    fi;
+  od;
+
+  for i in [1 .. Length(neighbours_v)] do
+    for j in [i + 1 .. Length(neighbours_v)] do
+      u := neighbours_v[i];
+      v := neighbours_v[j];
+      if not adjacency_function(v, u) then
```

Comment:
Probably the best you can do here is:
```suggestion
      if not IsDigraphEdge(D, v, u) then
```
Although I'm not totally sure.

But this change would let you get rid of the line 
```gap
adjacency_function := DigraphAdjacencyFunction(digraph);
```
above.

Reply:
Yep youre right, looking at the source, if the digraph does not have a particular adjacency function, then `DigraphAdjcencyFunction` actually just calls the `IsDigraphEdge` function under the hood, and otherwise, `IsDigraphEdge` does some clever checks to use a faster method if possible. So `IsDigraphEdge` should be best here and in the other place where `adjacency_function` was used.